dev-libs/libdbusmenu-qt/libdbusmenu-qt-0.9.3_pre20140619-r1.ebuild

```diff
@@ -0,0 +1,145 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+EBZR_REPO_URI="lp:libdbusmenu-qt"
+
+[[ ${PV} == 9999* ]] && BZR_ECLASS="bzr"
+inherit multibuild virtualx ${BZR_ECLASS} cmake-multilib
+
+DESCRIPTION="A library providing Qt implementation of DBusMenu specification"
+HOMEPAGE="https://launchpad.net/libdbusmenu-qt/"
+if [[ ${PV} == 9999* ]] ; then
+	KEYWORDS=""
+else
+	MY_PV=${PV/_pre/+14.10.}
+	SRC_URI="http://launchpad.net/ubuntu/+archive/primary/+files/${PN}_${MY_PV}.orig.tar.gz"
+	# upstream has no permissions to use some kde written code so repack git
+	# repo every time
+	#SRC_URI="http://dev.gentoo.org/~scarabeus/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
+	PATCHES=( "${FILESDIR}/${P}-optionaltests.patch" )
+fi
+
+LICENSE="LGPL-2"
+SLOT="0"
+IUSE="debug doc +qt4 qt5"
+S=${WORKDIR}/${PN}-${MY_PV}
+
+REQUIRED_USE="|| ( qt4 qt5 )"
+
+RDEPEND="
+	qt4? (
+		dev-qt/qtcore:4[${MULTILIB_USEDEP}]
+		dev-qt/qtdbus:4[${MULTILIB_USEDEP}]
+		dev-qt/qtgui:4[${MULTILIB_USEDEP}]
+	)
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtdbus:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+	)
+"
+DEPEND="${RDEPEND}
+	doc? ( app-doc/doxygen )
+	test? (
+		dev-libs/qjson[${MULTILIB_USEDEP}]
+		qt4? ( dev-qt/qttest:4[${MULTILIB_USEDEP}] )
+		qt5? ( dev-qt/qttest:5 )
+	)
+"
+
+DOCS=( NEWS README )
+
+# tests fail due to missing connection to dbus
+RESTRICT="test"
+
+pkg_setup() {
+	MULTIBUILD_VARIANTS=( $(usev qt4) $(usev qt5) )
+}
+
+src_prepare() {
+	[[ ${PV} == 9999* ]] && bzr_src_prepare
+	cmake-utils_src_prepare
+
+	use test || comment_add_subdirectory tests
+}
+
+# for qt4
+multilib_src_configure() {
+	local mycmakeargs=(
+		$(cmake-utils_use_with doc)
+		-DUSE_QT4=ON
+		-DQT_QMAKE_EXECUTABLE=/usr/$(get_libdir)/qt4/bin/qmake
+	)
+	cmake-utils_src_configure
+}
+
+# for qt5
+my_qt5_src_configure() {
+	local mycmakeargs=(
+		$(cmake-utils_use_with doc)
+		-DUSE_QT5=ON
+	)
+	cmake-utils_src_configure
+}
+
+src_configure() {
+	myconfigure() {
+		if [[ ${MULTIBUILD_VARIANT} = qt4 ]] ; then
+			multilib-minimal_src_configure
```

Comment:
It ain't pretty when you call `multilib_minimal_*` once, and `cmake-multilib_*` the other time :).